src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/OutgoingMessageProducer.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.producer;
+
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.model.ChipsKafkaMessage;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+@Component
+public class OutgoingMessageProducer implements MessageProducer {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss");
+
+    private ApplicationLogger logger;
+
+    private AvroSerializer avroSerializer;
+
+    private CHKafkaProducer producer;
+
+    @Autowired
+    @Qualifier("chips-kafka-send")
+    private Schema schema;
+
+    @Value("${kafka.retry.topic}")
+    private String retryTopicName;
+
+    @Autowired
+    public OutgoingMessageProducer(ApplicationLogger logger,
+                                   AvroSerializer avroSerializer,
+                                   CHKafkaProducer producer){
+        this.logger = logger;
+        this.avroSerializer = avroSerializer;
+        this.producer = producer;
+    }
+
+
+    @Override
+    public void writeToQueue(ChipsKafkaMessage chipsMessage) throws ServiceException {
+        try {
+            Message kafkaMessage = new Message();
+            byte[] serializedData = avroSerializer.serialize(chipsMessage, schema);
+            kafkaMessage.setValue(serializedData);
+            kafkaMessage.setTopic(retryTopicName);
+            kafkaMessage.setTimestamp(LocalDateTime.parse(chipsMessage.getCreatedAt(),
```

Comment:
I think it would be better to send the timestamp as a String version of epoch seconds. This is what i've done for the sending of the message from the strike-off-objections-api. This will hopefully eliminate any confusion when we have multiple client apps sending us messages trying to get the time format correct. By using the seconds, we can control the format we want to use inside this consumer app, we can convert the seconds into a localDateTime and the client apps don't have to worry about sending us messages with the correct time format.

Reply:
It looks like it was done this way (seconds) for emails in the api as well - https://github.com/companieshouse/strike-off-objections-api/blob/master/src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaEmailClient.java#L44

Reply:
As this requires a long would it be better to make the field a long rather than a string. Can you show me how you have done this?

Reply:
I would just pass the timestamp from the ChipsKafkaMessage into kafkaMessage - just use
            kafkaMessage.setTimestamp(Long.valueOf(chipsMessage.getCreatedAt()));

Reply:
we'll be replacing the ChipsKafkaMessage with the kafka model anyway, Ian is working on this.

Reply:
Replaced this.